api/api.go

```diff
@@ -219,6 +220,7 @@ func New(
 		volumePath:            defaultVolumesPath,
 		volumePathPermissions: defaultVolumesPathPermissions,
 		verboseLogging:        VerboseType(verboseLogging),
+		isiAuthType:           isiAuthType,
```

Comment:
add a check to make sure that the input auth type is one of 0 or 1, for other cases return with an error.